crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -283,6 +267,14 @@ impl SolverTimeout {
     }
 }
 
+impl TryFrom<time::Deadline> for SolverTimeout {
```

Comment:
nit: I don't recall the full discussion, but I remember that in the past (I believe Valentin) suggested refraining from using the conversion traits, when the conversion is highly specific to our use case (ie both types are defined by us and only ever need to be converted in this concrete case).

I actually think the right trait here would be From<Duration> and then created the timeout with `deadline.solver?.into()` to not burry the logic that the solver specific part of the deadline is used in some conversion trait.

Reply:
I personally like that the TryFrom forces the SolverTimeout to always have positive duration (not to depend on what the caller provided), but I am fine with your approach also. Fixed.